### PR TITLE
Feature/cancel pair

### PR DESCRIPTION
**Pull request by uv-orbs**
cancel multiple order reuses new cancel impl
cancel considers the symbol

### CODE REVIEW

File: data/redisrepo/get_orders.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			logctx.Error(ctx, "failed to parse order ID", logger.String("orderId", orderIdStr), logger.Error(err))
+			return nil, fmt.Errorf("failed to parse order ID: %v", err)
+		}
+		orderIds = append(orderIds, orderId)
+	}
+	return orderIds, nil
+
+}
+
+// func (r *redisRepository) GetOpenOrders(ctx context.Context, userId uuid.UUID, symbol models.Symbol) ([]models.Order, error) {
```
Comment (Luke-Rogerson) [inline data/redisrepo/get_orders.go:41]: Is this needed?

File: data/redisrepo/tx_order.go [hunk 2]
```diff
@@ -200,6 +200,7 @@ func (r *redisRepository) txStart(ctx context.Context) uint {
 	txid := r.ixIndex
 	r.txMap[txid] = tx
 
+	logctx.Info(ctx, "redisRepository txStart", logger.Int("txid", int(txid)))
```
Comment (Luke-Rogerson) [inline data/redisrepo/tx_order.go:203]: Probably should be a debug log

File: service/cancel_orders_for_user.go [hunk 3]
```diff
@@ -2,28 +2,41 @@ package service
...
 		logctx.Info(ctx, "no orders found for user", logger.String("userId", userId.String()))
 		return []uuid.UUID{}, err
 	}
-
-	if err != nil {
-		logctx.Error(ctx, "could not cancel orders for user", logger.Error(err), logger.String("userId", userId.String()))
-		return []uuid.UUID{}, fmt.Errorf("could not cancel orders for user: %w", err)
+	res := []uuid.UUID{}
+	for _, id := range ids {
+		order, err := s.getOrder(ctx, false, id)
```
Comment (Luke-Rogerson) [inline service/cancel_orders_for_user.go:21]: Use `FindOrdersByIds` here to only do one request to the DB

File: service/cancel_orders_for_user.go [hunk 4]
```diff
@@ -2,28 +2,41 @@ package service
...
+		if err != nil {
+			logctx.Error(ctx, "could not get order", logger.Error(err), logger.String("orderId", id.String()))
+		}
+		if order != nil {
+			// matching symbol only if provided
+			if symbol == "" || order.Symbol == symbol {
```
Comment (Luke-Rogerson) [inline service/cancel_orders_for_user.go:27]: Does Go let you pass empty string for ```symbol```? It's typed as ```symbol models.Symbol``` on line 12

File: service/cancel_orders_for_user_test.go [hunk 5]
```diff
@@ -29,19 +29,19 @@ func TestService_CancelOrdersForUser(t *testing.T) {
...
 
-		orderIds, err := s.CancelOrdersForUser(ctx, mocks.UserId)
+		orderIds, err := s.CancelOrdersForUser(ctx, mocks.UserId, "")
 		assert.Empty(t, orderIds)
 		assert.ErrorIs(t, err, models.ErrNotFound)
 	})
 
-	t.Run("should return error on unexpected error", func(t *testing.T) {
-		store := &mocks.MockOrderBookStore{User: nil, Error: assert.AnError}
+	// t.Run("should return error on unexpected error", func(t *testing.T) {
```
Comment (Luke-Rogerson) [inline service/cancel_orders_for_user_test.go:37]: Can this be updated?

File: data/redisrepo/get_orders.go (hunk 1 above)
Comment (uv-orbs) [reply data/redisrepo/get_orders.go:41]: in some point yes, but i didn't mean to leave it in.. thanks

File: service/cancel_orders_for_user.go (hunk 3 above)
Comment (uv-orbs) [reply service/cancel_orders_for_user.go:21]: done!

File: service/cancel_orders_for_user.go (hunk 4 above)
Comment (uv-orbs) [reply service/cancel_orders_for_user.go:27]: yes, it does

File: service/cancel_orders_for_user_test.go (hunk 5 above)
Comment (uv-orbs) [reply service/cancel_orders_for_user_test.go:37]: together

**Pull request merged**